### PR TITLE
test: use inplace 'rizin' binary for 'rzpipe' test

### LINKED ISSUE #1192: `rzpipe` test always fails on a system without `rizin` binary
Observed as a test failure as part of testsuite run on Gentoo distribution:

```
>>> Test phase: dev-util/rizin-0.2.1
meson test -C /tmp/portage/dev-util/rizin-0.2.1/work/rizin-0.2.1-build --num-processes 16
ninja: Entering directory `/tmp/portage/dev-util/rizin-0.2.1/work/rizin-0.2.1-build'
...
57/78 rzpipe                       FAIL            0.04s   exit status 1
>>> MALLOC_PERTURB_=23 /tmp/portage/dev-util/rizin-0.2.1/work/rizin-0.2.1-build/test/unit/test_rzpipe
...
```

This seems to happen because test assumes that `rizin` binary is available in $PATH. It is not when test is ran on a fresh system:

rizin-v0.2.1/test/unit/test_rzpipe.c:

```c
static bool test_rzpipe(void) {
#ifndef __WINDOWS__
        RzPipe *r = rzpipe_open("rizin -q0 -"); // <<<--- HERE
        mu_assert("rzpipe can spawn", r);
        char *hello = rzpipe_cmd(r, "?e hello world");
        mu_assert_streq(hello, "hello world\n", "rzpipe hello world");
        free(hello);
        rzpipe_close(r);
#else
        mu_test_status = MU_TEST_BROKEN;
#endif
        mu_end;
}
```

### PR DESCRIPTION
**Pull request by trofi**
Before the change test failed at the attempt of executing
'rizin' binary not present in $PATH:

```
>>> Test phase: dev-util/rizin-0.2.1
meson test
57/78 rzpipe                       FAIL            0.04s   exit status 1
>>> MALLOC_PERTURB_=23 test/unit/test_rzpipe
```

The change passes absolute path to the `rizin` binary.

Closes: https://github.com/rizinorg/rizin/issues/1192
Signed-off-by: Sergei Trofimovich <slyfox@gentoo.org>

 

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)

### CODE REVIEW

File: test/unit/meson.build [hunk 1]
```diff
@@ -94,7 +101,8 @@ if get_option('enable_tests')
 
   foreach test : tests
     exe = executable('test_@0@'.format(test), 'test_@0@.c'.format(test),
-      include_directories: [platform_inc],
+      # '.' to find autogenerated "test_config.h"
```
Comment (ret2libc) [inline test/unit/meson.build:104]: I think the addition of `.` is not necessary, as it is automatically added by meson AFAIR.

**Review by ret2libc — COMMENTED** [review]
Apart from one comment, it looks good! Thanks for working on this

File: test/unit/test_config.h.in [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#ifndef _TEST_CONFIG_H_
+#define _TEST_CONFIG_H_
+
+#define RIZIN_BUILD_PATH @RIZIN_BUILD_PATH@
+
+#endif _TEST_CONFIG_H_
```
Comment (XVilka) [inline test/unit/test_config.h.in]: Forgotten comment here:
```
 In file included from ../test/unit/test_rzpipe.c:7:
test/unit/test_config.h:6:8: error: extra tokens at end of #endif directive [-Werror=endif-labels]
    6 | #endif _TEST_CONFIG_H_
      |        ^~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

**Review by XVilka — CHANGES_REQUESTED** [review]

File: test/unit/test_config.h.in (hunk 2 above)
Comment (wargio) [inline test/unit/test_config.h.in]:
```suggestion
#endif /* _TEST_CONFIG_H_ */
```

File: test/unit/meson.build (hunk 1 above)
Comment (trofi) [reply test/unit/meson.build:104]: I added `.` as the build did not work as is:

```
FAILED: test/unit/test_rzpipe.p/test_rzpipe.c.o
ccache clang -Itest/unit/test_rzpipe.p -I. -I.. -I../librz/include -Isubprojects/sdb -I../subprojects/sdb -Isubprojects/sdb/src -I../subprojects/sdb/src -I../shlr/spp -Ishlr -I../shlr -I../librz/asm/arch/include -I../librz/asm/arch -I../librz/asm/arch/h8300 -I../librz/asm/arch/hexagon -I../librz/asm/arch/msp430 -I../librz/asm/arch/rsp -I../librz/asm/arch/mcore -I../librz/asm/arch/v850 -I../librz/asm/arch/propeller -I../librz/asm/arch/ebc -I../librz/asm/arch/cr16 -I../librz/asm/arch/8051 -I../librz/asm/arch/v810 -I../librz/asm/arch/or1k -I../librz/analysis/arch/gb -I../librz/analysis/arch/hexagon -I../librz/bin/format/elf -I../librz/bin/mangling -I../librz/bin/format -I../shlr/heap/include -Xclang -fcolor-diagnostics -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -O2 -g --std=gnu99 -D_GNU_SOURCE -Werror=sizeof-pointer-memaccess -fvisibility=hidden -MD -MQ test/unit/test_rzpipe.p/test_rzpipe.c.o -MF test/unit/test_rzpipe.p/test_rzpipe.c.o.d -o test/unit/test_rzpipe.p/test_rzpipe.c.o -c ../test/unit/test_rzpipe.c
../test/unit/test_rzpipe.c:7:10: fatal error: 'test_config.h' file not found
#include "test_config.h"
         ^~~~~~~~~~~~~~~
```

This target uses `implicit_include_directories: false` a few lines below. We can either remove that or add `.`. Which one is preferable?

**Review by ret2libc — APPROVED** [review]
Ok with . ! Thanks

File: test/unit/meson.build (hunk 1 above)
Comment (wargio) [reply test/unit/meson.build:104]: implicit_include_directories: false is necessary

**Review by wargio — APPROVED** [review]

**Pull request merged**